### DISCUSSION TITLE
Bevy Rapier 2D issue

### DISCUSSION THREAD

**Discussion by tgreener** [bevyengine/bevy#9119] [Q&A] [answered]
Hey friends, I have a problem I can't figure out.

I'm making a derpy little zelda clone. My character has a rigid body and a collider, and I'm trying to spawn a sword entity when the player presses the sword button. What I'm doing is creating the sword entity as a child entity of the character so that its transform remains connected to the player as the sword animates.

The sprite animation and visual motion works fine. The character moves using a Velocity component, and the sword sprite moves around the character the way I expect it would, but the sword collider (a sensor right now, but I've tried a kinematic position based rigid body too) DOESN'T move. It just stays in the initial position I put it. I've been adding the sensor/collider to the same entity as the sword sprite, so I'd assume they would transform the same.

What do?

Diagram of the entity structure:

Character: Transform, TextureAtlasSprite, RigidBody, Collider, Velocity, etc
|
-Weapon: SpatialBundle, etc
    |
    -Sword: TextureAtlasSprite, Sensor, SpatialBundle, etc

**Comment by WooterTheTroubleshooter** [comment, marked as answer]
I think this might be related: https://github.com/dimforge/bevy_rapier/issues/292#issuecomment-1446878738

**Reply by tgreener** [reply]
That does look like the exact same problem (even almost the exact same use case). Thanks for looking that up for me!
